libs/url/builder.ts

```diff
@@ -98,7 +107,7 @@ const buildURL = function (opts: FinalUrlOptions): string {
 
     var urlSignature = getSignature({
       privateKey: opts.privateKey,
-      url: intermediateURL,
+      url: encodeStringIfRequired(intermediateURL),
```

Comment:
This is wrong as explained on the call to @ankur-dwivedi . This encodes the query parameter (if present) again which are already encoded generating the wrong signature. Keep this logic consistent with the [Python SDK MR](https://github.com/imagekit-developer/imagekit-python/pull/63#pullrequestreview-1905912569) . 
Also add test cases where we have

-  a query parameter with diacritic
-  transformation such as `h-100,w-100` with `transformationPosition: "query"`

 to verify this change.

Reply:
updated the logic and have added the mentioned test cases